### PR TITLE
[FIX] mail: break loops when redirecting bounces

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
@std-odoo Trying to break loops when all emails are redirected to catchall using email forwarding (email redirected creates a bounce, that is redirected to catchall, that creates a bounce, that is redirected to catchall, ...) . What dya think ?

File: addons/test_mail/tests/test_mail_gateway.py [hunk 1]
```diff
@@ -292,6 +292,92 @@ def test_alias_mixin_copy_content(self):
         self.assertEqual(record_copy2.alias_bounced_content, new_content)
 
 
+@tagged('mail_gateway', 'prout')
```
Comment (tde-banana-odoo) [inline addons/test_mail/tests/test_mail_gateway.py:295]: This is temporary (and this is a reminder it should be temporary).

**Comment by std-odoo** [comment]
@tde-banana-odoo Hi :)

Does it preserve the return-path in that case ? 

Maybe we can set a special message id when we bounce and catch it in the references ?

A bit like we did with `loop-detection-bounce-email`, we can even make it more generic, and move that code
```py
# Add a reference with a tag, to be able to ignore response to this email
references = (
    message_dict.get('message_id', '') + ' '
    + tools.generate_tracking_message_id('loop-detection-bounce-email')
)
```

inside `_routing_create_bounce_email`, what do you think ?

**Comment by tde-banana-odoo** [comment]
Continued in https://github.com/odoo/odoo/pull/187696 .

**Pull request closed without merging**